### ISSUE TITLE
Better error message for Vuex mapActions typo

### ISSUE THREAD

**Issue by choonggg** [vuejs/vuex#1327]
### What problem does this feature solve?
While working using Vuex, I came across an error which took me quite a chunk of time to debug and I thought this deserves a better error messages.

While using map actions `...mapActions('select'Filter)` - notice the mistake with no square brackets in `Listings -> Filters -> SelectFilter` component. The error went up to top level component and appears as below.
```
vue.runtime.esm.js:587 [Vue warn]: Failed to resolve async component: function Listings() {
      return __webpack_require__.e/* import() */(0).then(__webpack_require__.bind(null, /*! @/pages/Listings/index.vue */ "./app/javascript/pages/Listings/index.vue"));
    }
Reason: TypeError: Cannot convert undefined or null to object
```

### What does the proposed API look like?
Show shows error message pointing to correct file and line.

**Comment by winniehell** [comment]
similar issue for `mapState`: https://github.com/vuejs/vuex/issues/1162

**Comment by ktsn** [comment]
fixed via #1297
